integration/apps/arithmetic_intensity/arithmetic_intensity.py

```diff
@@ -78,17 +83,23 @@ def create_appconf(mach, args):
     '''
 
     app_args = []
-    for arg in ['slowdown', 'base_internal_iterations', 'slow_ranks', 'floats', 'verbose', 'single_precision', 'list', 'iterations', 'benchmarks', 'start_time']:
+    for arg in ['slowdown', 'base_internal_iterations', 'slow_ranks', 'floats', 'verbose',
+                'single_precision', 'list', 'iterations', 'benchmarks', 'start_time']:
         values = vars(args)[arg]
         if values is not None:
             arg = "--" + arg.replace('_', '-')
-            app_args.append(arg)
-            if type(values) == list:
+            if isinstance(values, list):
+                app_args.append(arg)
                 app_args += [str(ii) for ii in values]
+            elif isinstance(values, bool):
+                if values is True:
+                    app_args.append(arg)
             else:
+                app_args.append(arg)
                 app_args.append(str(values))
```

Comment:
This problem is handled differently by geopmlaunch: https://github.com/geopm/geopm/blob/dev/scripts/geopmpy/launcher.py#L156

In that case, they only call add_argument for geopm-targeted args and not for any other wrapped args. I guess that would mean registering only --run-type and --ranks-per-node here. The rest are handled by forwarding the second return  value from parser.parse_known_args().

I think the current approach is also fine if that's preferred, particularly if you're interested in keeping all of the help text visible from the python wrapper.

Reply:
You are absolutely right. The only reason I did this, is to have the help text. Since I have added the run-type argument, to enable the help text from python I had to add all the program arguments to the python wrapper. Alternatively we can remove all that and add a '--app-help' option which we can pass to the benchmark command line as -h. Would you prefer that?

Reply:
We can even get fancier and override the argparse help to call the app help in addition to the help it provides.

Reply:
Actually, I realized this is something we need to do. The new options you added to the benchmark break if we use the tagged version.

Reply:
The new options add but don't change. Basically, they should only break in the sense that they don't exist if you have the old version, but you can still completely use all features of the old version even if you're aware of the features of the new version. The same applies whether you manually write the help text or delegate it to the installed version of the workload.

I think either approach can work (unless by your last comment, you mean something you tried to use isn't working). So I'd say go with whichever approach provides the least resistance to your experimentation workflow.